src/transformer/events/core/badge_awarded.php

```diff
@@ -0,0 +1,125 @@
+    $lang = utils\get_course_lang($course);
+    $badge = $repo->read_record_by_id('badge', $event->objectid);
+    $other = unserialize($event->other);
+    $issuedid = $other['badgeissuedid'];
+    $manual = $repo->read_record_by_id('badge_manual_award', $issuedid);
+    $awarder = $manual ? (utils\get_user($config, $repo->read_record_by_id('user', $manual->issuerid))) : 'System';
+    $badgetype = [1 => "Global", 2 => "Course"][$badge->type];
+    $course =  $badge->courseid ? $repo->read_record_by_id('course', $badge->courseid) : null;
+    
+    $statement = [[
+        'actor' => utils\get_user($config, $recipient),
+        'verb' => ['id' => 'https://w3id.org/xapi/tla/verbs/achieved',
+                   'display' => [
+                       'en-US' => 'Achieved'
+                   ]],
+        'object' => ['id' =>  $config['app_url'].'/badges/overview.php?id='.$event->objectid,
+                     'objectType' => 'Activity',
+                     'definition' => [
+                         'name' => [$lang => $badge->name],
+                         'description' => ['en-US' => $badge->description],
+                         'type' => 'https://xapi.edlm/profiles/edlm-lms/concepts/activity-types/badge',
+                         'extensions' => [
+                             'https://xapi.edlm/profiles/edlm-lms/v1/concepts/activity-extensions/badge-type' =>  $badgetype,
+                             'https://xapi.edlm/profiles/edlm-lms/v1/concepts/activity-extensions/badge-version' => $badge->version                             
+                         ]
+                     ],
+        ],
+        'result' => [
+            'success' => true,
+            'response' => $badge->message
+                ],
+        'context' => [
+            'language' => $lang,
+            'instructor' => $awarder,
+            'contextActivities' =>  [
+                'category' => ['id' => 'https://lms.edlm',
+                               'objectType' => 'Activity',
+                               'definition' => [
+                                   'name' => [
+                                       'en-US' => 'EDLM Moodle LMS'
+                                   ],
+                                   'type' => 'http://id.tincanapi.com/activitytype/lms'
+                               ]
+                ],
+            ],
+            'extensions' => [
+                'https://xapi.edlm/profiles/edlm-lms/v1/concepts/context-extensions/badge-assignment-method' => ($manual ? 'Manual' : 'Automatic'),
+                 'https://xapi.edlm/profiles/edlm-lms/concepts/context-extensions/lms-event-info' => [
+                     'http://moodle.org' => $CFG->release,
+                     'event_name' => '\\core\\event\\badge_awarded',
+                     'event_function' => '\\src\\transformer\\events\\core\\badge_awarded',
+                     'https://github.com/xAPI-vle/moodle-logstore_xapi' => ""
+                 ]
```

Comment:
This is something that should be created in a util ns (which already exists) and then applied to all statements, not put in here as static.

Reply:
`/src/transformer/utils/extensions/info.php`